client/src/lib/dependency-injection/applyFormMiddleware.js

```diff
@@ -28,27 +31,11 @@ const applyFormMiddleware = (reducer) => () => (state, action) => {
   }
 
   const schemaState = schema.state;
-  const manager = new SchemaStateManager(schemaState);
   let newState = {
     ...reducedState,
   };
-
-  const updates = formSchemaMiddleware(formState.values, manager);
-  newState = setIn(newState, `formSchemas.${schemaKey}.state`, {
-    ...schemaState,
-    fields: schemaState.fields.map((field) => {
-      const update = updates[field.name];
-
-      if (typeof update === 'undefined') {
-        return field;
-      }
-
-      return {
-        ...field,
-        value: update,
-      };
-    }),
-  });
+  const updates = formSchemaMiddleware(formState.values, schemaState);
+  newState = setIn(newState, `formSchemas.${schemaKey}.state`, updates);
```

Comment:
do we need a `if (updates)` check for this, in case a rogue middleware registered and forgot to return the state?